wear/src/main/java/com/example/wear/snippets/voiceinput/VoiceInputScreen.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.wear.snippets.voiceinput
+
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.content.Intent
+import android.speech.RecognizerIntent
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.example.wear.R
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults
+import com.google.android.horologist.compose.layout.ScalingLazyColumnDefaults.ItemType
+import com.google.android.horologist.compose.layout.ScreenScaffold
+import com.google.android.horologist.compose.material.Chip
+import com.google.android.horologist.compose.rotaryinput.rotaryWithScroll
+
+/**
+ * Shows voice input option
+ */
+@OptIn(ExperimentalHorologistApi::class)
+@Composable
+fun VoiceInputScreen(
+) {
+    // [START android_wear_voice_input]
+    var textForVoiceInput by remember { mutableStateOf("") }
+
+    val voiceLauncher =
+        rememberLauncherForActivityResult(
+            ActivityResultContracts.StartActivityForResult()
+        ) {
+            // This is where you process the intent and extract the speech text from the intent.
+            it.data?.let { data ->
+                val results = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
+                textForVoiceInput = results?.get(0) ?: "None"
+            }
+        }
+    val scrollState = rememberScrollState()
+
+    ScreenScaffold(scrollState = scrollState) {
```

Comment:
For the purposes of this snippet, is it possible to create something that doesn't involve including the Scaffold and Column code but still illustrates the point? In the same way that the existing snippet for voice doesn't include UI code? (Just thinking that this gets a bit long).

Reply:
good point, added a couple of `START_EXCLUDE android_wear_voice_input]` and `[END_EXCLUDE android_wear_voice_input]` to remove some code

Reply:
I dont think you need the android_wear_voice_input tag, you can just use start and end_exclude